code/modules/mob/living/silicon/pai/software.dm

```diff
@@ -285,15 +260,26 @@
 
 			if("doorjack")
 				if(href_list["jack"])
-					if(cable && cable.machine)
-						hackdoor = cable.machine
+					if(hacking_cable?.machine)
+						hackdoor = hacking_cable.machine
 						hackloop()
 				if(href_list["cancel"])
 					hackdoor = null
 				if(href_list["cable"])
-					var/turf/T = get_turf(loc)
-					cable = new /obj/item/pai_cable(T)
-					T.visible_message("<span class='warning'>A port on [src] opens to reveal [cable], which promptly falls to the floor.</span>", "<span class='hear'>You hear the soft click of something light and hard falling to the ground.</span>")
+					if(!QDELETED(hacking_cable))
+						qdel(hacking_cable) //clear any old cables
```

Comment:
What's the point of the `QDELETED` check here? Doesn't it not do anything for objects already queued?